src/server/scripts/Zandalar/AtalDazar/boss_priestess_alun_za.cpp

```diff
@@ -0,0 +1,684 @@
+                break;
+            }
+            default:
+                break;
+        }
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        if (!UpdateVictim())
+            return;
+
+        events.Update(diff);
+
+        if (me->HasUnitState(UNIT_STATE_CASTING))
+            return;
+
+        while (uint32 eventId = events.ExecuteEvent())
+        {
+            ExecuteEvent(eventId);
+            if (me->HasUnitState(UNIT_STATE_CASTING))
+                return;
+        }
+    }
+};
+
+// 130738 - Corrupted Gold
+struct npc_priestess_alun_za_corrupted_gold : public ScriptedAI
+{
+    npc_priestess_alun_za_corrupted_gold(Creature* creature) : ScriptedAI(creature) { }
+
+    void JustAppeared() override
+    {
+        DoCastSelf(SPELL_CORRUPTED_GOLD);
+
+        // TODO: orbs have collision but dont work properly so they are despawning when reach fixed position
+        float dist = 116.0f;
+        float randomAngle = frand(me->GetOrientation() - float(M_PI) / 2.0f, me->GetOrientation() + float(M_PI) / 2.0f);
+        float destX = me->GetPositionX() + dist * std::cos(randomAngle);
+        float destY = me->GetPositionY() + dist * std::sin(randomAngle);
+        float destZ = me->GetPositionZ();
+        Position pos(destX, destY, destZ);
+        me->GetMotionMaster()->MovePoint(POINT_DESPAWN, pos, true, {}, 2.0f, {}, {});
+    }
+
+    void MovementInform(uint32 type, uint32 pointId) override
+    {
+        if (type != POINT_MOTION_TYPE)
+            return;
+
+        if (pointId == POINT_DESPAWN)
+            me->DespawnOrUnsummon();
+    }
+
+private:
+    TaskScheduler _scheduler;
+};
+
+// 131009 - Spirit of Gold
+struct npc_priestess_alun_za_spirit_of_gold : public ScriptedAI
+{
+    npc_priestess_alun_za_spirit_of_gold(Creature* creature) : ScriptedAI(creature) { }
+
+    void JustAppeared() override
+    {
+        DoCastSelf(SPELL_SPAWN);
+        DoCastSelf(SPELL_TIKI_MASK);
+
+        _events.ScheduleEvent(EVENT_CHECK_TAINTED_BLOOD, 500ms);
+    }
+
+    void MovementInform(uint32 type, uint32 pointId) override
+    {
+        if (type != POINT_MOTION_TYPE)
+            return;
+
+        if (pointId == POINT_POOL)
+        {
+            DoCastSelf(SPELL_CORRUPT);
+            _events.ScheduleEvent(EVENT_CHECK_TAINTED_BLOOD, 1s);
+        }
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        UpdateVictim();
+
+        _events.Update(diff);
+
+        if (me->HasUnitState(UNIT_STATE_CASTING))
+            return;
+
+        while (uint32 eventId = _events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+                case EVENT_CHECK_TAINTED_BLOOD:
+                {
+                    // TODO: the npc should select the nearest at of tainted blood and go to his position, if not ats found should attack players atm is not working, just attacking
+                    std::vector<AreaTrigger*> atList = me->GetAreaTriggers(SPELL_TAINTED_BLOOD_AT);
```

Comment:
It doesn't work because GetAreaTriggers returns areatriggers owned by the unit using the getter, these ats are cast by the boss, just use the boss as the getter and it'll be fixed.